sql/src/main/java/io/crate/operation/projectors/ColumnIndexWriterProjector.java

```diff
@@ -148,18 +141,24 @@ public Result setNextRow(Row row) {
 
     @Override
     public void finish(RepeatHandle repeatHandle) {
+        BulkProcessorFutureCallback bulkProcessorFutureCallback = new BulkProcessorFutureCallback(failed, downstream);
```

Comment:
they're related as currently we still create the projector and then switch to the Consumer/BatchIterator pattern if possible. 
The problem with the callback was that this was still triggered even if the switch to the consumer was made - resulting in both the RowReciever & the Consumer being used.